ts/features/walletV3/payment/screens/WalletPaymentConfirmScreen.tsx

```diff
@@ -52,51 +95,179 @@ const WalletPaymentConfirmScreen = () => {
     }
   }, [authorizationUrlPot, navigation]);
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  const taxFee = pipe(
+    selectedPspOption,
+    O.chain(psp => O.fromNullable(psp.taxPayerFee)),
+    O.getOrElse(() => 0)
+  );
+
+  const paymentAmount = pipe(
+    pot.toUndefined(paymentDetailsPot),
+    O.fromNullable,
+    O.chain(paymentDetails => O.fromNullable(paymentDetails.amount)),
+    O.getOrElse(() => 0)
+  );
+
+  const paymentMethodDetails = pipe(
+    selectedMethodOption,
+    O.chain(method => O.fromNullable(method.details)),
+    O.toUndefined
+  );
+
+  const totalAmount = paymentAmount + taxFee;
+
+  const selectedMethod = O.toUndefined(selectedMethodOption);
+
+  const selectedPsp = O.toUndefined(selectedPspOption);
+
   const handleStartPaymentAuthorization = () => {
-    dispatch(
-      walletPaymentAuthorization.request({
-        paymentAmount: 1000 as AmountEuroCents,
-        paymentFees: 1000 as AmountEuroCents,
-        pspId: "A",
-        transactionId: "A",
-        walletId: "A"
-      })
-    );
+    if (selectedMethod && selectedPsp && pot.isSome(transactionPot)) {
```

Comment:
since you need all three of them to correctly render the screen, what do you think of using a `SequenceS` (like in `BeneficiaryDetailsScreen`, line 51)?
then you could get a map with the three variables and fold it to either:
- undef -> render loading
- def -> you have all variables, so you can just render the screen

Reply:
sorry, wrong line 😓 
this was intended for line 142

Reply:
Thanks, now the code looks a way better!

Addressed into 57ef65abca625e11601100e5b40c4848f23f0e14